spec/integration/token_authentication_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'spec_helper'
+
+describe 'integration' do
+  include_context 'with a generated app'
+
+  it 'returns a forbidden error' do
+    get "http://localhost:#{app_port}/"
+    expect_json(error: 'Forbidden')
+  end
+
+  context 'with correct token query params' do
+    it 'returns the root presenter' do
+      get "http://localhost:#{app_port}/?token=replace-me"
+      expect_json_keys(:_links)
+    end
+  end
+
+  context 'with correct token query params' do
+    it 'returns the root presenter' do
+      get "http://localhost:#{app_port}/", 'Http-Authorization' => 'replace-me'
```

Comment:
Very basic specs proving that a newly generated app will behave as expected with the token authentication helper.